ansible_collections/arista/cvp/plugins/module_utils/resources/schemas/v3.py

```diff
@@ -245,3 +245,384 @@
+                                "value": "value2"
+                            }
+                        ]
+                    },
+                    {
+                        "interface": "Ethernet1/2",
+                        "tags": [
+                            {
+                                "name": "tag1",
+                                "value": "value1"
+                            },
+                            {
+                                "name": "tag2",
+                                "value": "value2"
+                            }
+                        ]
+                    }
+                ]
+            }
+        ]
+    ],
+    "additionalItems": True,
+    "items": {
+        "$id": "#/items",
+        "anyOf": [
+            {
+                "$id": "#/items/anyOf/0",
+                "type": "object",
+                "title": "The first anyOf schema",
+                "description": "An explanation about the purpose of this instance.",
+                "default": {},
+                "examples": [
+                    {
+                        "device": "leaf1",
+                        "device_tags": [
+                            {
+                                "name": "tag1",
+                                "value": "value1"
+                            },
+                            {
+                                "name": "tag2",
+                                "value": "value2"
+                            }
+                        ],
+                        "interface_tags": [
+                            {
+                                "interface": "Ethernet1/1",
+                                "tags": [
+                                    {
+                                        "name": "tag1",
+                                        "value": "value1"
+                                    },
+                                    {
+                                        "name": "tag2",
+                                        "value": "value2"
+                                    }
+                                ]
+                            },
+                            {
+                                "interface": "Ethernet1/2",
+                                "tags": [
+                                    {
+                                        "name": "tag1",
+                                        "value": "value1"
+                                    },
+                                    {
+                                        "name": "tag2",
+                                        "value": "value2"
+                                    }
+                                ]
+                            }
+                        ]
+                    }
+                ],
+                "required": [
+                    "device",
+                    "device_tags",
+                    "interface_tags"
+                ],
```

Comment:
not sure if this is the right way or not, but removing that part I was able to add just device tags or just interface tags while maintaining the schema checking functionality, for instance

```
---
- name: Test cv_tag_v3
  hosts: cv_server
  connection: local
  gather_facts: no
  vars:
    CVP_TAGS:
      - device: tp-avd-leaf1
        device_tags:
          - name1: tamas
            value: test1
          - name1: tamas2
            value: test2
  tasks:
    - name: "create tags"
      arista.cvp.cv_tag_v3:
        tags: "{{CVP_TAGS}}"
        mode: create
        state: assign
        auto_create: true
```

that create the two device tags `tamas:test1` and `tamas2:test2`

when modifying one of the keys e.g.:

```
    CVP_TAGS:
      - device: tp-avd-leaf1
        device_tags:
          - name1: tamas1
            value: test1
          - name1: tamas21
            value: test21
```

that expectedly threw an error:

fatal: [cv_server]: FAILED! => changed=false
  msg: |-
    Error, your input is not valid against current schema:
     {'device': 'tp-avd-leaf1', 'device_tags': [{'name1': 'tamas1', 'value': 'test1'}, {'name1': 'tamas21', 'value': 'test21'}]}

likewise I was able to create interface tags only and when messing with the schema vars it was complaining as expected:

```
        interface_tags:
          - interface: Ethernet1
            tags:
              - name1: tag111
                value: value111
              - name: tag221
                value: value221
          - interface: Ethernet2
            tags:
              - name: tag113
                value1: value111
              - name: tag223
                value: value223
```

```
fatal: [cv_server]: FAILED! => changed=false
  msg: |-
    Error, your input is not valid against current schema:
     {'device': 'tp-avd-leaf1', 'interface_tags': [{'interface': 'Ethernet1', 'tags': [{'name1': 'tag111', 'value': 'value111'}, {'name': 'tag221', 'value': 'value221'}]}, {'interface': 'Ethernet2', 'tags': [{'name': 'tag113', 'value1': 'value111'}, {'name': 'tag223', 'value': 'value223'}]}]}
```

Reply:
The recent commit should have this change. Do test again.

Reply:
Indeed, it works as expected now, thanks!